app/scripts/controllers/permissions/background-api.js

```diff
@@ -38,23 +43,39 @@ export function getPermissionBackgroundApiMethods({
   };
 
   // To add more than one account when already connected to the dapp
-  const addMoreAccounts = (origin, accounts) => {
+  const addMoreAccounts = (origin, addresses) => {
     const caip25Caveat = getCaip25Caveat(origin);
     if (!caip25Caveat) {
       throw new Error(
         `Cannot add account permissions for origin "${origin}": no permission currently exists for this origin.`,
       );
     }
 
-    const ethAccounts = getEthAccounts(caip25Caveat.value);
+    const internalAccounts = addresses.map((address) => {
+      return accountsController.getAccountByAddress(address);
+    });
+
+    // Only the first scope in the scopes array is needed because
+    // setPermittedAccounts currently sets accounts on all matching
+    // namespaces, not just the exact CaipChainId.
+    const caipAccountIds = internalAccounts.map((internalAccount) => {
+      const { namespace, reference } = parseCaipChainId(
```

Comment:
Do we need to parse here? We can probably just prefix the scope directly

Reply:
yes indeed: https://github.com/MetaMask/metamask-extension/pull/29887/commits/c6f7968d85ae56c7f35159f0f7aff9e30b4e4ca3